### PR TITLE
added metrics messages to DTCEventVerifier

### PR DESCRIPTION
**Pull request by gianipez**

### CODE REVIEW

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc [hunk 1]
```diff
@@ -102,13 +105,21 @@ bool mu2e::DTCEventVerifier::filter(art::Event& event)
...
-      std::cout << "[ArtFragmentsFromDTCEvents::produce] Found nHandlesnFragments  " 
+      std::cout << "[ArtFragmentsFromDTCEvents::produce] Found nFragments  " 
 		<< fragments.size() <<std::endl;
     }
-  
+  if (metricMan != nullptr)
+    {
+      std::ostringstream oss;
+      oss << "Found nFragments=  " << fragments.size();
+      metricMan->sendMetric("nFragments", oss.str(), "xml_string",
```
Comment (eflumerf) [inline artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: Probably better to just do:
`metricMan->sendMetric("nFragments", fragments.size(), "Fragments", metrics_reporting_level_, artdaq::MetricMode::LastPoint);`
In general, numeric metrics should be preferred over string metrics where possible.

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc [hunk 2]
```diff
@@ -140,9 +151,30 @@ bool mu2e::DTCEventVerifier::filter(art::Event& event)
...
 	if ( subEvtWTag != evtWTag) { evtHeader->ewt_check = 0; } //different EWT in a subEvent
 	
+	if (metricMan != nullptr && diagLevel_ > 10)
+	  {
+	    std::ostringstream oss;
+	    oss << "SubEvent: "     << &subEvt - &data.GetSubEvents()[0] 
+		<< ", DTC_ID = "    << dtcID 
+		<< ", EWT_CHECK = " << ( (subEvtWTag != evtWTag) ? 0 : 1)
+		<< ", DTC_CHECK = " << ( ((dtcs_.insert(dtcID).second) && (!isFirstEvent_)) ? 0 : 1);
+	    metricMan->sendMetric("SubEventSnapshot", oss.str(), "xml_string",
```
Comment (eflumerf) [inline artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: Again, consider using numeric metrics. Metric collection is very fast, so don't worry about sending multiple metrics.

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc [hunk 3]
```diff
@@ -140,9 +151,30 @@ bool mu2e::DTCEventVerifier::filter(art::Event& event)
...
+	
 	//if (subHeader->dtc_mac == dtcHeader->dtc_mac){ evtDTC_ID = dtcID;}
       }//end loop over the subevents
     
+    if (metricMan != nullptr)
+      {
+	std::ostringstream oss;
+	oss << "Fragment: " << &frag - &fragments[0]
+	    << ", EWT_CHECK = " << evtHeader->ewt_check 
+	    << ", DTCs = "      << evtNDTCs << "/" << nDTCs_
```
Comment (eflumerf) [inline artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: You can send metrics for both evtNDTCs and evtNDTCs/nDTCs_, again, metric collection is very fast and you can send as many as you like.

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc (hunk 1 above)
Comment (gianipez) [reply artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: thanks for the suggestion! I made the changes as you suggested.

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc (hunk 3 above)
Comment (gianipez) [reply artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: done

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc (hunk 2 above)
Comment (gianipez) [reply artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: done

File: artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc [hunk 4]
```diff
@@ -140,9 +149,34 @@ bool mu2e::DTCEventVerifier::filter(art::Event& event)
       
 	if ( subEvtWTag != evtWTag) { evtHeader->ewt_check = 0; } //different EWT in a subEvent
 	
+	if (metricMan != nullptr && diagLevel_ > 10)
+	  {
+	    metricMan->sendMetric("SubEventID"      , int(&subEvt - &data.GetSubEvents()[0]), "SubEvent",
```
Comment (eflumerf) [inline artdaq-mu2e/ArtModules/DTCEventVerifier_module.cc]: This line is super concerning...you're doing pointer arithmetic?

**Review by eflumerf — APPROVED** [review]
LGTM

**Pull request merged**